azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/util/DeploymentTemplateBuilder.java

```diff
@@ -76,16 +75,14 @@
    private final String group;
    private final Template template;
    private final Json json;
+   private final LoginCredentials loginCredentials;
 
    private AzureTemplateOptions options;
    private Iterable<String> tags;
    private Map<String, String> userMetaData;
    private List<ResourceDefinition> resources;
    private Map<String, String> variables;
-   private static String loginUser;
-   private static String loginPassword;
```

Comment:
This was a source of race conditions: different calls to the adapter with different credentials were overriding this shared value. Shared mutable state is evil, specially in static variables.